tutorial/opt/aero/plot_optHist.py

```diff
@@ -3,49 +3,67 @@
-axes[0].autoscale(enable=True, tight=True)
+        # Objective Optimality and Feasibility
+        axes[4].plot("nMajor", "optimality", data=hist_values, label="Optimality")
+        axes[4].plot("nMajor", "feasibility", data=hist_values, label="Feasibility")
+        axes[4].set_yscale("log")
+        axes[4].axhline(1e-4, linestyle="--", color="gray")
+        axes[4].annotate("Convergence Criteria", xy=(3, 9e-5), ha="left", va="top", fontsize=24, color="gray")
+        axes[4].legend(fontsize=20, labelcolor="linecolor", loc="upper right", frameon=False)
+        axes[4].autoscale(enable=True, tight=True)
 
-axes[1].plot("nMajor", "obj", data=histValues, label="Objective")
-axes[1].set_ylabel("Objective \n($C_D$)", rotation="horizontal", ha="right", fontsize=24)
+    else:
+        _, axes = plt.subplots(nrows=4, sharex=True, figsize=(10, 14))
 
-axes[2].plot("nMajor", "alpha_fc", data=histValues, label="Alpha")
-axes[2].set_ylabel(r"$\alpha [^\circ]$", rotation="horizontal", ha="right", fontsize=24)
+    # Objective (Drag Coefficient)
+    axes[0].plot("iter", "obj", data=hist_values, label="Objective ($C_D$)")
+    axes[0].set_ylabel("Objective ($C_D$)", rotation="horizontal", ha="right", fontsize=24)
 
-axes[3].plot("nMajor", "twist", data=histValues, label="Twist")
-axes[3].set_ylabel(r"Twist $[^\circ]$", rotation="horizontal", ha="right", fontsize=24)
+    # Angle of Attack
+    axes[1].plot("iter", "alpha_wing", data=hist_values, label="Angle of Attack")
+    axes[1].set_ylabel(r"$\alpha$ [$^\circ$]", rotation="horizontal", ha="right", fontsize=24)
 
-axes[4].plot("nMajor", "cl_con_fc", data=histValues, label="cl_con")
-axes[4].set_ylabel("Lift constraint", rotation="horizontal", ha="right", fontsize=24)
+    # Wing Twist
+    axes[2].plot("iter", "twist", data=hist_values, label="Wing Twist")
+    axes[2].set_ylabel("Twist [$^\circ$]", rotation="horizontal", ha="right", fontsize=24)
 
-for ax in axes:
-    ax.spines["right"].set_visible(False)
-    ax.spines["top"].set_visible(False)
+    # Lift Constraint
+    axes[3].plot("iter", "cl_con_wing", data=hist_values, label="Lift Constraint")
+    axes[3].set_ylabel("Lift Constraint", rotation="horizontal", ha="right", fontsize=24)
 
-    # Drop the rest of the spines
-    ax.spines["left"].set_position(("outward", 12))
-    ax.spines["bottom"].set_position(("outward", 12))
+    # Adjust subplot aesthetics
+    for ax in axes:
+        ax.spines["right"].set_visible(False)
+        ax.spines["top"].set_visible(False)
+        ax.spines["left"].set_position(("outward", 12))
+        ax.spines["bottom"].set_position(("outward", 12))
 
-plt.savefig(os.path.join(args.outputDir, "aero_wing_opt_hist.png"))
+    plt.show()
+    plt.savefig(os.path.join(args.outputDir, "aero_wing_opt_hist.jpg"), bbox_inches="tight")
+
+
+if __name__ == "__main__":
+    # plt.rcParams["text.usetex"] = True  # Comment out if latex installation is not present
+    plt.rcParams["font.family"] = "serif"
```

Comment:
I generally prefer having all `plt` commands before any function definition, at the module level. Not blocking this PR though, since this is probably personal preference.